src/internal/m/m.go

```diff
@@ -0,0 +1,441 @@
+// Package m implements lightweight metrics.
+
+// A metric is a name, a set of key/value pairs (set by
+// the underlying logger), and a value.  A value can be an absolute value ("gauge"), an incremental
+// value ("counter"), or a set of samples (like "histogram", though lossless in this
+// implementation).  Metrics are most useful when additional code analyzes the entire log of a
+// program's run; this is called "analysis code" or "anaylsis software" throughout the
+// documentation.
+//
+// Values are stored by writing the set of operations to build the value to the logs.  For example,
```

Comment:
See my comment about a `MetricSpace`  I think this section describes a particular implementation of that and doesn't need to be included in the mental model of the metrics system.